Support codegen --opt-ionvar-copy in coreneuron recipe

cattabiani opened the pull request:
CLI Nmodl options are passed to coreneuron during spack install.
There is a new option: codegen --opt-ionvar-copy=TRUE/FALSE.

New way to set that option to false (true by default):
spack install coreneuron@develop~codegenopt

cattabiani committed:
Support codegen --opt-ionvar-copy in coreneuron recipe

CLI Nmodl options are passed to coreneuron during spack install.
There is a new option: codegen --opt-ionvar-copy=TRUE/FALSE.

New way to set that option to false (true by default):
spack install coreneuron@develop~codegenopt

cattabiani committed:
squash! change whitespaces for flake8 in CI

pramodk commented on var/spack/repos/builtin/packages/coreneuron/package.py:
```diff
@@ -72,12 +73,13 @@ class Coreneuron(CMakePackage):
     conflicts('+sympyopt', when='~sympy')
     conflicts('+sympy', when='~nmodl')
     conflicts('+sympy', when='coreneuron@0.17')  # issue with include directories
+    conflicts('~codegenopt', when='~nmodl')
```
why this is a conflict ? I think we can simply remove this.

cattabiani replied:
it does not make sense to call +/~codegenopt without nmodl. It is an nmodl option...

pramodk replied:
yeah thats true..but also its "legal" to have `~codegenopt` when `~nmodl` and hence was saying we can remove it. But anyway this is very minor thing, we can leave as it is.

cattabiani replied:
in this case it is  "legal" to have `+sympy` when `~nmodl` but we raise a conflict. In addition, if you wrote `~codegenopt` with `~nmodl` you wanted to do something else. Better to fail as soon as possible

pramodk commented on var/spack/repos/builtin/packages/coreneuron/package.py:
```diff
@@ -153,6 +155,9 @@ def get_cmake_args(self):
 
         nmodl_options = 'codegen --force'
 
+        if spec.satisfies('~codegenopt'):
```
Don’t we need to pass true flag explicitly? I am thinking now that the codegenopt should be false on nmodl side by default (if it’s not the case by default).

cattabiani replied:
I wanted to remain as close as possible to the current state. It was always true before. Now, if we remove it, we do something. A little bit like ~report I think

pramodk replied:
I am thinking of making it false default because with the current setting naive user would get incorrect current traces. Hence making optimization flag optional would make more sense?

cattabiani replied:
That is true indeed. In this case I need to change nmodl too for standard cmake build

cattabiani commented:
please retest

cattabiani committed:
coreneuron variant codegenopt default=FALSE

switched from default=TRUE because naive users could get
false traces with coreneuron

matz-e reviewed (approved):
Sorry… got a bit delayed.

Seems all good?
